packages/eui/.storybook/addons/code-snippet/decorators/utils.ts

```diff
@@ -0,0 +1,218 @@
+    node.props?.__EMOTION_TYPE_PLEASE_DO_NOT_USE__ != null
+  ) {
+    const { __EMOTION_TYPE_PLEASE_DO_NOT_USE__: emotionData } = node.props;
+    const isForwardRefComponent = isForwardRef(emotionData);
+
+    // we need to rely here on the reference Emotion stores to know what component this actually is
+    const replacementName = isForwardRefComponent
+      ? emotionData.__docgenInfo.displayName
+      : typeof emotionData === 'string'
+      ? emotionData
+      : emotionData?.displayName;
+
+    // remove internal component underscore markings
+    return replacementName ? replacementName.replace('_', '') : displayName;
+  }
+
+  return displayName;
+};
+
+export const getComponentDisplayName = (
+  context: StoryContext | undefined
+): string | undefined => {
+  if (!context) return undefined;
+  const component = context.component as ComponentType<any> & {
+    __docgenInfo?: { displayName?: string };
+  };
+
+  return component?.displayName ?? component.__docgenInfo?.displayName;
+};
+
+export const isEmotionComponent = (node: ReactElement<any>): boolean => {
+  const displayName = getElementDisplayName(node);
+  const matches =
+    typeof displayName === 'string'
+      ? displayName.match(/^(Emotion)(\w)*/g)
+      : null;
+
+  return matches != null;
+};
+
+/* Story specific checks */
+export const isStoryComponent = (
+  node: ReactElement<any>,
+  context: StoryContext | undefined
+): boolean => {
+  if (!context) return false;
+
+  const displayName = getEmotionComponentDisplayName(node);
+
+  return displayName === context?.component?.displayName;
+};
+
+/**
+ * checks if the outer most component is a parent of the actual story component
+ */
+export const isStoryParent = (
+  node: ReactElement<any>,
+  context: StoryContext | undefined
+): boolean => {
+  if (!context) return false;
+
+  const displayName = getEmotionComponentDisplayName(node);
+
+  if (!displayName) return false;
+
+  const parentComponents = context.title.split('/');
+  parentComponents.shift();
+  parentComponents.pop();
+
+  return parentComponents.includes(displayName);
+};
+
+export const isSubcomponent = (
+  node: ReactElement<any>,
+  context: StoryContext | undefined
+): boolean => {
+  if (!context) return false;
+
+  const parentComponents = context.title.split('/');
+  parentComponents.shift();
+  const displayName = getEmotionComponentDisplayName(node);
+
+  if (!displayName || !parentComponents || parentComponents.length === 0) {
+    return false;
+  }
+
+  let isSub = false;
+
+  for (const parent of parentComponents) {
+    if (typeof displayName === 'string' && displayName.includes(parent)) {
+      isSub = true;
+    }
+  }
+
+  return isSub;
+};
+
+export const isStatefulComponent = (node: ReactElement<any>): boolean => {
+  const displayName = getEmotionComponentDisplayName(node);
+  const isStateful =
+    typeof displayName === 'string'
+      ? displayName.match(/^(Stateful|Component)(\w)*/)
+      : null;
```

Comment:
Not sure if this feels more readable to you?
```suggestion
  const isStateful = typeof displayName === 'string' && (displayName.startsWith('Stateful') || displayName.startsWith('Component'));
```

Reply:
Updated in this [commit](https://github.com/elastic/eui/pull/7716/commits/118adc8eed0f4a85d7eac509d9c32fa65d666e2d)